### DISCUSSION TITLE
Does hono have an OpenAPI endpoint like FastAPI?

### DISCUSSION THREAD

**Discussion by janzheng** [honojs/hono#1574] [Q&A] [answered]
Hi there,

I couldn't find a resource for this, but does hono have something like what FastAPI has for generating an JSON Schema for the API endpoints, like: https://fastapi.tiangolo.com/tutorial/first-steps/?h=openapi#check-the-openapijson?

**Comment by ccssmnn** [comment, marked as answer]
There is a guide on OpenAPI [here](https://hono.dev/guides/openapi). With the packages mentioned there, you can use the `OpenAPIHono`-Class to create an OpenAPI App and serve a Swagger UI.

**Reply by janzheng** [reply]
This is the link I used (the given is 404'd): https://github.com/honojs/middleware/tree/main/packages/zod-openapi

It's a bit more boilerplate-y than FastAPI but it works! Thanks so much.

**Reply by iatomic1** [reply]
Yh i am also coming from fastapi and it just works on fastapi but u have to write more code to make it work here, apart from that i'm thinking of using Hono in my side projects

**Comment by MathurAditya724** [comment]
Hey 👋🏼 , I created this package https://github.com/rhinobase/hono-openapi, which can reduce the boilerplate code and supports multiple validation libs. Let me know if you face any issues, it's still in beta but the response seems to be good
